lib/autoupdate/agent/updater.go

```diff
@@ -341,18 +353,53 @@ func (u *Updater) Install(ctx context.Context, override OverrideConfig) error {
 	// Only write the configuration file if the initial update succeeds.
 	// Note: skip_version is never set on failed enable, only failed update.
 
-	if err := writeConfig(u.ConfigPath, cfg); err != nil {
+	if err := writeConfig(u.UpdateConfigPath, cfg); err != nil {
 		return trace.Wrap(err, "failed to write %s", updateConfigName)
 	}
 	u.Log.InfoContext(ctx, "Configuration updated.")
 	return trace.Wrap(u.notices(ctx))
 }
 
+// findAgentProxy finds a proxy in teleport.yaml if not specified or set in update configuration.
+// Note that any implicitly defaulted port in teleport.yaml is explicitly defaulted (to 3080) by this method.
+func (u *Updater) findAgentProxy(ctx context.Context) string {
+	fc, err := config.ReadFromFile(u.TeleportConfigPath)
```

Comment:
I don't know if I'm very comfortable with parsing the full teleport config as would break in case of incompatible change or too large version mismatch.

On one hand this is not a large issue if this opportunistic lookup fails, on the other, it would be way more resilient to unmarshal into a small struct only containing the fields we are looking for.